src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -561,7 +561,14 @@ internal HttpRequestMessage BuildSetTriggersRequest()
             var url = default(string);
             if (_environment.IsKubernetesManagedHosting())
             {
-                url = $"http://{ManagedKubernetesBuildServiceName}.{ManagedKubernetesBuildServiceNamespace}.svc.cluster.local:{ManagedKubernetesBuildServicePort}/operations/settriggers";
+                var buildServiceHostname =
+                    _environment.GetEnvironmentVariable(EnvironmentSettingNames.BuildServiceHostname);
+
```

Comment:
Move this to environment extensions

Reply:
Done. 
Could you have a look? @pragnagopa